### PR TITLE
m68k: initial port

### PR DESCRIPTION
**Pull request by netbsduser**
This is an initial port of mlibc to the m68k family, based on what is already working in Keyronex. Sysdeps are not included, but I have made an initial, not yet tested m68k port of the Linux sysdeps in another branch.
Some elements of the port are initially stubbed, viz. 68881-style FPU support and the setjmp family of functions.

### CODE REVIEW

File: options/internal/m68k-include/mlibc/thread.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#include <mlibc/internal-sysdeps.hpp>
+#include <mlibc/tcb.hpp>
+#include <bits/ensure.h>
+
+namespace mlibc {
+
+extern "C" void *__m68k_read_tp();
+
+inline Tcb *get_current_tcb() {
+	// On m68k, the TCB is below 0x7000 below the thread pointer.
```
Comment (64) [inline options/internal/m68k-include/mlibc/thread.hpp:13]: nit: below repeated twice

File: options/internal/m68k-include/mlibc/thread.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	void *ptr = __m68k_read_tp();
+	return reinterpret_cast<Tcb *>((uintptr_t)ptr - 0x7000 - sizeof(Tcb));
```
Comment (64) [inline options/internal/m68k-include/mlibc/thread.hpp:15]: can you use tlsOffsetFromTp?

Comment (netbsduser) [reply options/internal/m68k-include/mlibc/thread.hpp:15]: It's defined elsewhere, so I followed what the aarch64 port does: https://github.com/managarm/mlibc/blob/8c6e9be5b94e9162146217772c2d3dceb2376c7f/options/internal/aarch64-include/mlibc/thread.hpp#L8-L13
https://github.com/managarm/mlibc/blob/8c6e9be5b94e9162146217772c2d3dceb2376c7f/options/rtld/generic/linker.cpp#L40

File: options/internal/m68k-include/mlibc/thread.hpp (hunk 1 above)
Comment (netbsduser) [reply options/internal/m68k-include/mlibc/thread.hpp:13]: I worded this in a very ugly way, it was supposed to convey that
```
+-----+---------------|------+
| TCB |  0x7000 bytes | more |
+-----+---------------+------+
                      ^
                      TP
```
but could benefit from being reworded.

Comment (64) [inline options/internal/m68k-include/mlibc/thread.hpp:13]: I mean:
```suggestion
	// On m68k, the end of the TCB is 0x7000 below the thread pointer.
```

Comment (netbsduser) [reply options/internal/m68k-include/mlibc/thread.hpp]: That works, amended with that.

**Pull request closed without merging**